src/transformers/models/ict/modeling_ict.py

```diff
@@ -0,0 +1,863 @@
+# coding=utf-8
+# Copyright 2023 Authors at City University of Hong Kong, Microsoft Cloud + AI,
+# The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch ICT model."""
+
+
+import math
+from typing import Dict, List, Optional, Set, Tuple, Union
+
+import torch
+import torch.utils.checkpoint
+import torchvision.models as models
+from torch import nn
+
+from ...activations import ACT2FN
+from ...modeling_outputs import BaseModelOutput, BaseModelOutputWithPooling, MaskedImageCompletionOutput
+from ...modeling_utils import PreTrainedModel
+from ...pytorch_utils import find_pruneable_heads_and_indices, prune_linear_layer
+from ...utils import (
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    logging,
+    replace_return_docstrings,
+)
+from .configuration_ict import ICTConfig, ICTGuidedUpsamplerConfig, ICTTransformerConfig
+
+
+logger = logging.get_logger(__name__)
+
+# General docstring
+_CONFIG_FOR_DOC = "ICTConfig"
+
+# Base docstring
+_CHECKPOINT_FOR_DOC = "sheonhan/ict-imagenet-256"
+_EXPECTED_OUTPUT_SHAPE = [3, 256, 256]
+
+
+ICT_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "sheonhan/ict-imagenet-256",
+    "sheonhan/ict-ffhq-256",
+    "sheonhan/ict-places-256",
+    # See all ICT models at https://huggingface.co/models?filter=ict
+]
+
+
+class ICTEmbeddings(nn.Module):
```

Comment:
We're trying to make model names more consistent and so should all be camel-case now. Can you update ICT -> Ict everywhere? 

```suggestion
class IctEmbeddings(nn.Module):
```